lib/astarte_core/device/capabilities.ex

```diff
@@ -0,0 +1,45 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2025 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+defmodule Astarte.Core.Device.Capabilities do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias Astarte.Core.Device.Capabilities
+
+  @required_fields []
+
+  @permitted_fields [:purge_properties_compression_format] ++ @required_fields
+
+  @primary_key false
+  embedded_schema do
+    field :purge_properties_compression_format, Ecto.Enum,
+      values: [:zlib, :plaintext],
+      default: :zlib
+
+    field :device_id, Ecto.UUID, primary_key: true
```

Comment:
\> `@primary_key false`
\> `primary_key: true`

![image](https://tenor.com/view/think-emoji-thinking-in-thought-rotate-gif-8083088.gif)

Reply:
Jokes aside, this structure will be used only for validation, so no need for a relationship with the device ID. What we're interested in is just that an error should be provided if (at the moment) there is something different from `zlib` or `plaintext` in the input, therefore the previous version of the schema (i.e. without `device_id`) is enough